platform/default/mbgl/storage/offline_download.cpp

```diff
@@ -57,7 +57,14 @@ void tileCover(const OfflineRegionDefinition& definition, style::SourceType type
     for (uint8_t z = clampedZoomRange.min; z <= clampedZoomRange.max; z++) {
         definition.match(
                 [&](const OfflineTilePyramidRegionDefinition& reg){ tileCover(reg.bounds, z, fn); },
-                [&](const OfflineGeometryRegionDefinition& reg){ tileCover(reg.geometry, z, fn); }
+                [&](const OfflineGeometryRegionDefinition& reg){ tileCover(reg.geometry, z, fn); },
+                [&](const OfflineTileListRegionDefinition& reg){
+                    if(z == clampedZoomRange.min) {
```

Comment:
I'm new to mapbox / tile internals - can you help me understand why we apply the function to each tile when we're at min zoom?

Reply:
fn is actually just a function that adds an item to a an std::vector. The way this is written is totally dumb, it doesn't matter what the zoom is, just that it only happens one time. It would make more sense if this just wasnt in this zoom loop, but variant.match requires that all of the variant types have an entry.